### PR TITLE
[MS] Add style and time function on Device Page

### LINKED ISSUE #5700: Add a new device (style)
Edit the style according to the mockups

Image

### PR DESCRIPTION
**Pull request by fabienSvtr**
Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
- [x] Give a meaningful title to your PR
- [x] Ensure the commit history is sanitized
- [x] Describe your changes
- [x] Link any related issue in the description
- [x] Link any dependent pull request in the description

- Users are reminded to save their password. 
- The page design complies with the mock-ups.
- The user knows when each device has been added. 

closes #5700

### CODE REVIEW

File: client/src/common/date.ts [hunk 1]
```diff
@@ -3,19 +3,20 @@
 import { DateTime } from 'luxon';
 import { ComposerTranslation } from 'vue-i18n';
 
-export function formatTimeSince(date: DateTime | undefined, t: ComposerTranslation, d: any, defaultValue='', format='long'): string {
+export function formatTimeSince(date: DateTime | undefined, t: ComposerTranslation, d: any, defaultValue='', format:'long'|'short' = 'long',
```
Comment (Max-7) [inline client/src/common/date.ts:6]: Some inconsistencies in spaces that the linter apparently doesn't catch. The right format is `var: type = value` with those precise spaces.

File: client/src/common/injectionKeys.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import { DateTime } from 'luxon';
 
 export interface Formatters {
-  timeSince(date: DateTime | undefined, defaultValue?: string, format?: string): string;
+  timeSince(date: DateTime | undefined, defaultValue?: string, format?: 'long' | 'short', roundDays?:boolean): string;
```
Comment (Max-7) [inline client/src/common/injectionKeys.ts:6]: See my previous remark on spaces being inconsistent.

File: client/src/common/date.ts [hunk 3]
```diff
@@ -3,19 +3,20 @@
...
+  roundDays = false): string {
   if (!date) {
     return defaultValue;
   }
   // Get the difference in ms
   const diff = DateTime.now().diff(date, ['years', 'months', 'days', 'hours', 'minutes', 'seconds']).toObject();
 
   // More than 6 days, just display the date as is
-  if (!diff || diff.years && diff.years > 0 || diff.months && diff.months > 0 || diff.days && diff.days > 6) {
+  if (!diff || diff.years && diff.years > 0 || diff.months && diff.months > 0 || diff.days && diff.days > 6 && !roundDays) {
```
Comment (Max-7) [inline client/src/common/date.ts:15]: Wouldn't
```typescript
if (!diff || diff.years && diff.years > 0 || diff.months && diff.months > 0 || diff.days && diff.days > 6) {
  return d(date.toJSDate(), format);
} else if (diff.days && roundDays) {
  return t('common.date.lastLoginDays', {days: diff.days}, diff.days);
} else if (diff.days && diff.days > 0) {
  return t('common.date.lastLoginDays', {days: diff.days}, diff.days);
}
```
work and be easier to read?

Reading this I also realized that we need some tests for this function.

File: client/src/locales/fr-FR.json [hunk 4]
```diff
@@ -460,6 +460,21 @@
         "addDevice": "Ajouter",
         "noDevices": "Aucun appareil.",
         "activeDeviceBadge": "Actif",
+        "joinedOn": "Rejoint le",
```
Comment (Max-7) [inline client/src/locales/fr-FR.json:463]: `lastLoginDays` should probably be updated in this file too.

File: client/src/views/devices/DevicesPage.vue [hunk 5]
```diff
@@ -2,37 +2,141 @@
...
+            class="body-sm danger"
           >
-            <ion-icon
-              :icon="add"
-            />
-            {{ $t('DevicesPage.addDevice') }}
-          </ion-button>
+            {{ $t('DevicesPage.restorePassword.notDone.label') }}
+          </ion-label>
+          <div class="restore-password-header">
```
Comment (Max-7) [inline client/src/views/devices/DevicesPage.vue:65]: I see some new things but no tests...

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: client/src/common/date.ts (hunk 3 above)
Comment (fabienSvtr) [reply client/src/common/date.ts:15]: You mean added this function to the existing `if`? Because we still need `seconds`, `minutes` and `hours`.

**Review by Max-7 — APPROVED** [review]

**Review by NicoTuxx — CHANGES_REQUESTED** [review]
- [x] the date text is not really clear in french

image

File: client/src/locales/en-US.json [hunk 6]
```diff
@@ -460,6 +460,21 @@
         "addDevice": "Add",
         "noDevices": "No devices.",
         "activeDeviceBadge": "Current",
+        "joinedOn": "Joined on: {lastLoginDays}",
```
Comment (NicoTuxx) [inline client/src/locales/en-US.json]:
```suggestion
        "joinedOn": "Joined:",
```
Unused

File: client/src/locales/fr-FR.json (hunk 4 above)
Comment (NicoTuxx) [inline client/src/locales/fr-FR.json]:
```suggestion
        "joinedOn": "Rejoint :",
```

File: client/src/locales/fr-FR.json [hunk 7]
```diff
@@ -460,6 +460,21 @@
...
+        "restorePassword": {
+            "title": "Fichiers de récupération de mot de passe",
+            "done": {
+                "subtitle": "Vous avez sauvegardé les fichiers de récupération de mot de passe.",
+                "label": "Fichiers déjà téléchargés",
+                "button": "Re-télécharger les fichiers"
+            },
+            "notDone": {
+                "label": "Action requise",
+                "subtitle": "Parsec ne stocke pas votre mot de passe. ",
```
Comment (NicoTuxx) [inline client/src/locales/fr-FR.json:473]:
```suggestion
                "subtitle": "Parsec ne stocke pas votre mot de passe.",
```

File: client/src/main.ts [hunk 8]
```diff
@@ -114,9 +114,9 @@ async function setupApp(): Promise<void> {
     .use(i18n);
 
   app.provide(FormattersKey, {
-    'timeSince': (date: DateTime | undefined, defaultValue = '', format = 'long'): string => {
+    'timeSince': (date: DateTime | undefined, defaultValue = '', format: 'long' | 'short' = 'long', roundDays?:boolean): string => {
```
Comment (NicoTuxx) [inline client/src/main.ts:117]:
```suggestion
    'timeSince': (date: DateTime | undefined, defaultValue = '', format: 'long' | 'short' = 'long', roundDays?: boolean): string => {
```

File: client/src/views/devices/DevicesPage.vue [hunk 9]
```diff
@@ -2,37 +2,141 @@
...
+              </ion-item>
+            </ion-list>
+          </div>
+        </div>
+
+        <!-- restore password card -->
+        <!-- files not downloaded -->
+        <div
+          class="restore-password"
+          v-if="passwordSaved === false"
```
Comment (NicoTuxx) [inline client/src/views/devices/DevicesPage.vue:58]:
```suggestion
          v-if="!passwordSaved"
```

**Review by NicoTuxx — CHANGES_REQUESTED** [review]

File: client/src/common/date.ts [hunk 10]
```diff
@@ -3,7 +3,8 @@
 import { DateTime } from 'luxon';
 import { ComposerTranslation } from 'vue-i18n';
 
-export function formatTimeSince(date: DateTime | undefined, t: ComposerTranslation, d: any, defaultValue='', format='long'): string {
+export function formatTimeSince(date: DateTime | undefined, t: ComposerTranslation, d: any, defaultValue='',
```
[the remark this answers is not in the document]
Comment (fabienSvtr) [reply client/src/common/date.ts:6]: Done

File: client/src/common/date.ts [hunk 11]
```diff
@@ -3,7 +3,8 @@
...
+  format: 'long'|'short' = 'long', roundDays = false): string {
```
[the remark this answers is not in the document]
Comment (fabienSvtr) [reply client/src/common/date.ts:7]: Done

File: client/src/common/injectionKeys.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabienSvtr) [reply client/src/common/injectionKeys.ts:6]: Done

File: client/src/locales/fr-FR.json (hunk 7 above)
Comment (fabienSvtr) [reply client/src/locales/fr-FR.json:473]: Done

File: client/src/views/devices/DevicesPage.vue (hunk 9 above)
Comment (fabienSvtr) [reply client/src/views/devices/DevicesPage.vue:58]: Done

**Review by NicoTuxx — APPROVED** [review]

**Pull request merged**